src/org/labkey/test/params/FieldDefinition.java

```diff
@@ -836,4 +846,47 @@ public String getSecondRange()
         }
     }
 
+    /**
+     * TextChoice is implemented using a validator, however it is more 'limited' in scope. The user does not name a TextChoice
+     * validator or add a description or error message. A TextChoice is a lot like a look-up field, but it is not linked
+     * to an external data source. The user only provides the list of (string) values that the field will display in the dropdown.
+     * Another difference is that there can only be one TextChoice on a field, whereas you can have multiple validators on a field.
+     */
+    public static class TextChoiceValidator extends FieldValidator<TextChoiceValidator>
+    {
+        private List<String> _values;
+
+        public TextChoiceValidator(List<String> values)
+        {
+            // The TextChoice validator only has a name and no description or message.
+            // And the name is generated (not user defined).
+            super("Text Choice Validator", "", "");
+            _values = values;
+        }
```

Comment:
Consider locking down the collection of values.
```suggestion
        private final List<String> _values;

        public TextChoiceValidator(List<String> values)
        {
            // The TextChoice validator only has a name and no description or message.
            // And the name is generated (not user defined).
            super("Text Choice Validator", "", "");
            _values = Collections.unmodifiableList(values);
        }
```

Reply:
Done.